Remove use of #[may_dangle]

```rust
unsafe impl<#[may_dangle] K, #[may_dangle] V> Drop for RawTable<K, V> {
    fn drop(&mut self) {
        // ...
    }
}
```

Those `#[may_dangle]` attributes require activating two features, `#![feature(dropck_eyepatch)]` and `#![feature(generic_param_attrs)]`.

It compiles fine without `#[may_dangle]`, and passes tests, so it's not clear to me what this is doing.  If it's a point of safety, this could be a problem.  If it's just a matter of type flexibility, we can probably accept that resulting limitation without it.

Ah, the difference is in code like this:

```rust
    let (a, mut b) = (0, rayon_hash::HashSet::new());
    b.insert(&a);
```

Without `#[may_dangle]`, you get an error like:

```text
error[E0597]: `a` does not live long enough
 --> examples/may_dangle.rs:6:1
  |
5 |     b.insert(&a);
  |               - borrow occurs here
6 | }
  | ^ `a` dropped here while still borrowed
  |
  = note: values in a scope are dropped in the opposite order they are created
```

It's still fine if you declare `a` *before* `b`, just not simultaneously.

I think this is a limitation we'll just have to live with, lacking stable `#[may_dangle]`.

--- a/examples/may_dangle.rs
+++ b/examples/may_dangle.rs
@@ -1,0 +1,17 @@
+extern crate rayon_hash;
+
+fn main() {
+    // The standard collections can have type parameters with
+    // the same lifetime as the collection itself:
+    let (x, mut set) = (0, std::collections::HashSet::new());
+    set.insert(&x);
+
+    // FIXME: We can't do this without `#[may_dangle]` on `Drop for RawTable`:
+    // let (x, mut set) = (0, rayon_hash::HashSet::new());
+    // set.insert(&x);
+
+    // Instead, we our type parameters must strictly outlive the collection.
+    let x = 0;
+    let mut set = rayon_hash::HashSet::new();
+    set.insert(&x);
+}

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,8 +1,6 @@
 #![feature(alloc)]
 #![feature(allocator_api)]
-#![feature(dropck_eyepatch)]
 #![cfg_attr(rayon_hash_unstable, feature(fused))]
-#![feature(generic_param_attrs)]
 #![cfg_attr(rayon_hash_unstable, feature(placement_new_protocol))]
 #![feature(shared)]
 #![feature(unique)]

--- a/src/std_hash/table.rs
+++ b/src/std_hash/table.rs
@@ -1171,7 +1171,8 @@ impl<K: Clone, V: Clone> Clone for RawTable<K, V> {
     }
 }
 
-unsafe impl<#[may_dangle] K, #[may_dangle] V> Drop for RawTable<K, V> {
+// unsafe impl<#[may_dangle] K, #[may_dangle] V> Drop for RawTable<K, V> {
+impl<K, V> Drop for RawTable<K, V> {
     fn drop(&mut self) {
         if self.capacity() == 0 {
             return;

